openhands/llm/llm.py

```diff
@@ -64,8 +64,9 @@ def __init__(
         self.config = copy.deepcopy(config)
         self.metrics = metrics if metrics is not None else Metrics()
         self.cost_metric_supported = True
-        self.supports_prompt_caching = (
-            self.config.model in cache_prompting_supported_models
+        self.enable_prompt_caching = (
```

Comment:
I find this approach can lead to confusion as in the code we would have 2 differently named attributes and I'd be hard pressed to remember, which one came from the `config`.

For clearer readability I'd like to propose this approach:
Replace `self.enable_prompt_caching` with a method like `def is_caching_active(self) -> bool` on the LLM class level, which returns the result as in the 2 lines below.